utils/hct/hctdb.py

```diff
@@ -8080,6 +8080,11 @@ def build_valrules(self):
             "Invalid input record type for node launch type",
             "%0 node shader '%1' has incompatible input record type (should be %2)",
         )
+        self.add_valrule_msg(
+            "Decl.MeshNodeOutputRecord",
+            "A mesh shader may not have any output records,",
+            "Mesh node shader '%0' has unexpected output record '%1'",
```

Comment:
nit: I think the wording isn't right, though it might be fine for now.  I think the term we would want to use is "node outputs" rather than "output records".
Maybe:
```suggestion
            "Decl.NodeOutputOnMeshLaunch",
            "Node outputs are not allowed on mesh launch node shaders.",
            "node shader '%0' has disallowed node output '%1' for 'mesh' launch",
```